src/main/java/com/shopify/mappers/LegacyToFulfillmentOrderMapping.java

```diff
@@ -0,0 +1,170 @@
+package com.shopify.mappers;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.shopify.ShopifySdk;
+import com.shopify.model.ShopifyFulfillment;
+import com.shopify.model.ShopifyFulfillmentOrder;
+import com.shopify.model.ShopifyFulfillmentOrder.SupportedActions;
+import com.shopify.model.ShopifyFulfillmentOrderLineItem;
+import com.shopify.model.ShopifyFulfillmentOrderMoveLocationPayload;
+import com.shopify.model.ShopifyFulfillmentOrderMoveRequestRoot;
+import com.shopify.model.ShopifyFulfillmentOrderPayloadLineItem;
+import com.shopify.model.ShopifyFulfillmentPayload;
+import com.shopify.model.ShopifyFulfillmentPayloadRoot;
+import com.shopify.model.ShopifyLineItem;
+import com.shopify.model.ShopifyLineItemsByFulfillmentOrder;
+import com.shopify.model.ShopifyTrackingInfo;
+import com.shopify.model.ShopifyUpdateFulfillmentPayload;
+import com.shopify.model.ShopifyUpdateFulfillmentPayloadRoot;
+
+public class LegacyToFulfillmentOrderMapping {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(LegacyToFulfillmentOrderMapping.class);
+
+	/**
+	 * the idea here is to create a payload similar to what <a href=
+	 * "https://shopify.dev/docs/api/admin-rest/2023-04/resources/fulfillmentorder#post-fulfillment-orders-fulfillment-order-id-move">we
+	 * have here</a>, the resulting payload will be sent to shopify via the
+	 * active ShopifySdk instance
+	 * 
+	 * @see ShopifySdk
+	 * @see ShopifyFulfillmentOrderMoveRequestRoot
+	 * 
+	 * @param newLocation
+	 *            the new location that the fulfillment order will be moved to
+	 * @param fulfillmentOrder
+	 *            the fulfillment order to be moved
+	 * @return an ShopifyFulfillmentOrderMoveRequestRoot instance to be sent to
+	 *         shopify
+	 */
+	public static ShopifyFulfillmentOrderMoveRequestRoot toShopifyMoveFulfillmentOrder(final String newLocation,
+			ShopifyFulfillmentOrder fulfillmentOrder) {
+		try {
+			final ShopifyFulfillmentOrderMoveLocationPayload payload = new ShopifyFulfillmentOrderMoveLocationPayload();
+			payload.setNewLocationId(newLocation);
+			for (final ShopifyFulfillmentOrderLineItem fulfillmentOrderLineItem : fulfillmentOrder.getLineItems()) {
+				payload.getFulfillmentOrderLineItems().add(new ShopifyFulfillmentOrderPayloadLineItem(
+						fulfillmentOrderLineItem.getId(), fulfillmentOrderLineItem.getQuantity()));
+			}
+
+			ShopifyFulfillmentOrderMoveRequestRoot root = new ShopifyFulfillmentOrderMoveRequestRoot();
+			root.setFulfillmentOrder(payload);
+			return root;
+		} catch (final Exception e) {
+			LOGGER.error(
+					"There was an error parsing fulfillmentOrder into a ShopifyFulfillmentOrder moveRequest payload");
+			return null;
```

Comment:
i'd be in favor of bubbling the exception up. Can you also add the exception to the log?

Reply:
yep, good call on those.
Way more visibility when something happens!